collector/processor/batchprocessor/batch_processor.go

```diff
@@ -254,7 +326,35 @@ func (b *shard) resetTimer() {
 
 func (b *shard) sendItems(trigger trigger) {
 	sent, bytes, err := b.batch.export(b.exportCtx, b.processor.sendBatchMaxSize, b.processor.telemetry.detailed)
-	if err != nil {
+
+	numItemsBefore := b.totalSent
+	numItemsAfter := b.totalSent + sent
+	// The current batch can contain items from several different producers. Ensure each producer gets a response back.
+	for len(b.pending) > 0 && numItemsBefore < numItemsAfter {
+		// Waiter only had some items in the current batch
+		if numItemsBefore + b.pending[0].numItems > numItemsAfter {
+			b.pending[0].numItems -= (numItemsAfter - numItemsBefore)
+			b.pending[0].respCh <- partialError{err: err}
+			numItemsBefore = numItemsAfter
+		} else { // waiter gets a complete response.
+			numItemsBefore += b.pending[0].numItems
+			if trigger == triggerTimeout {
+				err = multierr.Append(err, errTimedOut)
```

Comment:
Hmm, this looks like we're making up an error, let's discuss. Is this necessary?

Reply:
Oh maybe I'm confused.. should the waiter be notified if their request times out? I thought the answer was yes, so that's why I'm letting the `completeError` indicate the timeout. Do you think timeout should simply be logged instead?